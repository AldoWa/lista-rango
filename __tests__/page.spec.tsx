import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import Page from '../app/page'
 
describe('Page', () => {
  it('Should render the first page', () => {
    render(<Page />)
 
    const heading = screen.getByRole('heading', { level: 1 })
 
    expect(heading).toBeInTheDocument()
    expect(heading).toHaveTextContent('Bem-vindo ao Lista Rango')

    const form = screen.getByTestId('home-form')
    expect(form).toBeInTheDocument()
  })

  it('Should match snapshot', () => {
    const { container } = render(<Page />)
    expect(container).toMatchSnapshot()
  })
})