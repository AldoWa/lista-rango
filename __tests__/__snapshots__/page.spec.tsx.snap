// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Should match snapshot 1`] = `
<div>
  <h1
    class="text-2xl text-center text-gray-800 mt-8"
  >
    Bem-vindo ao Lista Rango
  </h1>
  <div
    class="flex justify-center"
    data-testid="home-form"
  >
    <input
      class=" shadow-md px-8 w-full  rounded-full  bg-slate-100 h-10  focus:outline-none my-8 max-w-4xl "
      placeholder="Buscar estabelecimento"
      type="text"
      value="?search="
    />
  </div>
  <div
    class="grid grid-cols-3 gap-9"
    role="list"
  >
    <a
      aria-label="Go to the restaurant"
      class="shadow-md flex items-center  rounded relative  h-28 "
      data-testid="restaurant-card"
      href="/restaurants/1/menu"
    >
      <img
        alt="Restaurant Logo"
        class="h-28"
        data-nimg="1"
        decoding="async"
        height="112"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=256&q=100"
        srcset="/_next/image?url=%2Fimg.jpg&w=128&q=100 1x, /_next/image?url=%2Fimg.jpg&w=256&q=100 2x"
        style="color: transparent;"
        width="112"
      />
      <div
        class="ml-6"
      >
        <p
          class="text-base"
          data-testid="restaurant-name"
        >
          Restaurant 1
        </p>
        <p
          class="text-xs"
          data-testid="restaurant-address"
        >
          Address 1
        </p>
      </div>
      <div
        aria-live="polite"
        class="rounded-full
          w-14
          h-14
          flex
          items-center
          justify-center
          absolute
          -top-2 -right-4
          bg-[#B5ABD4]
        "
        role="status"
      >
        <span
          class="text-white text-xs leading-3 text-center font-bold"
          data-testid="restaurant-opened"
          role="alert"
        >
          Fechado
        </span>
      </div>
    </a>
    <a
      aria-label="Go to the restaurant"
      class="shadow-md flex items-center  rounded relative  h-28 "
      data-testid="restaurant-card"
      href="/restaurants/2/menu"
    >
      <img
        alt="Restaurant Logo"
        class="h-28"
        data-nimg="1"
        decoding="async"
        height="112"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=256&q=100"
        srcset="/_next/image?url=%2Fimg.jpg&w=128&q=100 1x, /_next/image?url=%2Fimg.jpg&w=256&q=100 2x"
        style="color: transparent;"
        width="112"
      />
      <div
        class="ml-6"
      >
        <p
          class="text-base"
          data-testid="restaurant-name"
        >
          Restaurant 2
        </p>
        <p
          class="text-xs"
          data-testid="restaurant-address"
        >
          Address 2
        </p>
      </div>
      <div
        aria-live="polite"
        class="rounded-full
          w-14
          h-14
          flex
          items-center
          justify-center
          absolute
          -top-2 -right-4
          bg-[#B5ABD4]
        "
        role="status"
      >
        <span
          class="text-white text-xs leading-3 text-center font-bold"
          data-testid="restaurant-opened"
          role="alert"
        >
          Fechado
        </span>
      </div>
    </a>
  </div>
</div>
`;
